helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -147,10 +147,16 @@ protected static RealmAwareZkClient createZkClient(HelixManager manager) {
     if (Boolean.getBoolean(SystemPropertyKeys.MULTI_ZK_ENABLED) || zkAddress == null) {
       String clusterName = manager.getClusterName();
       String shardingKey = HelixUtil.clusterNameToShardingKey(clusterName);
+      RealmAwareZkClient.RealmAwareZkConnectionConfig zkConnectionConfig =
```

Comment:
Could you explain why we don't just use zkConnectionConfig directly?

Reply:
You're right, we don't have to create a sharding key and rebuild config anymore. Changing.